repository_service_tuf/helpers/tuf.py

```diff
@@ -57,6 +78,181 @@ def to_dict(self):
         }
 
 
+class RootInfo:
+    _root_md: Metadata[Root]
+    root_keys: Dict[str, RSTUFKey]  # used to easily get all root keys
+    _current_signing_keys: Dict[str, RSTUFKey]  # required for signing
+    _newly_added_signing_keys: Dict[str, RSTUFKey]  # required for signing
+    online_key: RSTUFKey
+    _initial_root_md_obj: Metadata[Root]  # required to check for changes
+
+    @property
+    def threshold(self) -> int:
+        return self._root_md.signed.roles[Root.type].threshold
+
+    @threshold.setter
+    def threshold(self, value: int) -> None:
+        self._root_md.signed.roles[Root.type].threshold = value
+
+    @property
+    def expiration(self) -> datetime:
+        return self._root_md.signed.expires
+
+    @expiration.setter
+    def expiration(self, value: datetime) -> None:
+        self._root_md.signed.expires = value
+
+    @property
+    def expiration_str(self) -> str:
+        return f"{self.expiration.strftime('%Y-%b-%d')}"
+
+    @property
+    def signing_keys(self) -> List[RSTUFKey]:
+        return list(self._current_signing_keys.values()) + list(
+            self._newly_added_signing_keys.values()
+        )
+
+    @property
+    def keys_list(self) -> List[RSTUFKey]:
+        return list(self.root_keys.values())
+
+    def __init__(
+        self,
+        root_md: Metadata[Root],
+        root_keys: Dict[str, RSTUFKey],
+        online_key: RSTUFKey,
+    ):
+        self._root_md = root_md
+        self.root_keys = root_keys
+        self.online_key = online_key
+        self._current_signing_keys = {}
+        self._initial_root_md_obj = copy.deepcopy(self._root_md)
+        self._newly_added_signing_keys = {}
+
+    @classmethod
+    def _get_name(cls, key: Union[Key, RSTUFKey]) -> str:
+        name: str
+        if isinstance(key, Key):
+            name = key.keyid[:7]
+            if (
+                key.unrecognized_fields.get("name") is not None
+                and key.unrecognized_fields.get("name") != ""
+            ):
+                name = key.unrecognized_fields["name"]
+
+        elif isinstance(key, RSTUFKey):
+            if key.name is None or key.name == "":
+                name = key.key["keyid"][:7]
+            else:
+                name = key.name
+
+        return name
+
+    @classmethod
+    def from_md(cls, root_md: Metadata[Root]) -> "RootInfo":
+        root_keys: Dict[str, RSTUFKey] = {}
+        for keyid in root_md.signed.roles[Root.type].keyids:
+            tuf_key: Key = root_md.signed.keys[keyid]
+            name = cls._get_name(tuf_key)
+            tuf_key.unrecognized_fields["name"] = name
```

Comment:
Yes, it is. 
We want to have `names` for all root keys even if the user doesn't provide one.
This helps later when removing a key as the user gives us a name to the key he wants to remove and even if the key doesn't have a name initially we want to give a name to the key, so the user can easily write down when he wants to remove it.
Additionally, when we have to remove the key later we can store the keys in a dict where the keys are the name of the corresponding root keys.

Finally, this won't invalidate the check if `root` was changed because we copy into `_initial_root_md_obj` as the final step of the initialization after all keys have names added to them.